src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -320,17 +306,15 @@ bool ObjectMonitor::enter_is_async_deflating() {
   return false;
 }
 
-void ObjectMonitor::enter_for_with_contention_mark(JavaThread* locking_thread, ObjectMonitorContentionMark& contention_mark) {
-  // Used by ObjectSynchronizer::enter_for to enter for another thread.
+bool ObjectMonitor::enterI_with_contention_mark(JavaThread* locking_thread, ObjectMonitorContentionMark& contention_mark) {
+  // Used by ObjectSynchronizer::enter_for() to enter for another thread.
   // The monitor is private to or already owned by locking_thread which must be suspended.
   // So this code may only contend with deflation.
   assert(locking_thread == Thread::current() || locking_thread->is_obj_deopt_suspend(), "must be");
```

Comment:
I feel like the comments and assert now belong in `ObjectMonitor::enter_for_with_contention_mark`. 

`enterI_with_contention_mark` should be renamed. This is now a sort of `TryLock_with_contention_mark`. 

`add_to_contentions(1);` below could be changed to `contention_mark.extend()`.

Reply:
fixed